news/579.bugfix

```diff
@@ -0,0 +1 @@
+Prevent assignment of List[NoneType] or Dict[..., NoneType] to a ListConfig/DictConfig whose values are typed as non-optional.
```

Comment:
Doesn't it prevent assignment of None values to a List Config with a non optional element type?

Reply:
I think you're right.

Reply:
Hmm... It turns out that `list_cfg.append(None)` was not as expected. I just added another failing test...

Reply:
Ok, should be cleared up now. The news fragment is improved in https://github.com/omry/omegaconf/pull/696/commits/00322883311ded5ca76ed87c4f46ef8c6ca5ca34.

> Hmm... It turns out that `list_cfg.append(None)` was not as expected.

Nevermind about this, I was confused because I didn't realize that `cfg = ListConfig([], element_type=int)` has `is_optional(cfg)==True` by default.